tidy3d/components/tcad/doping.py

```diff
@@ -70,11 +70,30 @@ def _get_indices_in_box(self, coords: dict, meshgrid: bool = True):
 
         return indices_in_box, X, Y, Z, normal_axis
 
+    @pd.root_validator(skip_on_failure=True)
+    def check_dimensions(cls, values):
+        """Make sure dimensionality is specified correctly. I.e.,
+        a 2D box must be defined with an inf size in the normal direction."""
+
+        size = values["size"]
+        for dim in range(3):
+            if size[dim] == 0:
+                dim_name = "xyz"[dim]
+                raise SetupError(
+                    f"The doping box has been setup with 0 size in the {dim_name} direction. "
+                    "If this was intended to be a 2D box, the box must have inf size in the "
+                    "perpendicular direction."
+                )
+
+        return values
+
 
 class ConstantDoping(AbstractDopingBox):
     """
     Sets constant doping :math:`N` in the specified box with a :parameter`size` and :parameter:`concentration`.
 
+    For 2D behavior, the box must have infinite size in the homogenous direction.
```

Comment:
Same here and below, instead of `2D behavior` maybe `translationally invariant`?